integration-tests/validation.py

```diff
@@ -579,12 +593,67 @@ def validate_status_messages(model):
     )
 
     # Create registry
-    test_registry = os.path.join(here, 'templates', 'test-registry', 'test-registry.yaml')
-    await scp_to(test_registry, vessel, '/tmp/test-registry')
-    await kubectl('create -f /tmp/test-registry/test-registry.yaml')
-    output = await kubectl('get svc test-registry -o json')
-    data = json.loads(output)
-    registry_port = data['spec']['ports'][0]['nodePort']
+    nodes = await kubectl_get('nodes')
+    node_hostname = nodes['items'][0]['metadata']['labels']['kubernetes.io/hostname']
+    await kubectl_create({
+        'apiVersion': 'v1',
+        'kind': 'Pod',
+        'metadata': {
+            'name': 'test-registry',
+            'labels': {
+                'app': 'test-registry'
+            }
+        },
+        'spec': {
+            'containers': [{
+                'name': 'registry',
+                'image': 'registry:2.6.2',
+                'ports': [{
+                    'containerPort': 5000,
+                    'protocol': 'TCP'
+                }],
+                'env': [
+                    {'name': 'REGISTRY_AUTH_HTPASSWD_REALM', 'value': 'test-registry'},
+                    {'name': 'REGISTRY_AUTH_HTPASSWD_PATH', 'value': '/test-registry/htpasswd'},
+                    {'name': 'REGISTRY_HTTP_TLS_KEY', 'value': '/test-registry/tls.key'},
+                    {'name': 'REGISTRY_HTTP_TLS_CERTIFICATE', 'value': '/test-registry/tls.crt'}
+                ],
+                'volumeMounts': [{
+                    'name': 'test-registry',
+                    'mountPath': '/test-registry'
+                }]
+            }],
+            'volumes': [{
+                'name': 'test-registry',
+                'secret': {
+                    'secretName': 'test-registry'
+                }
+            }],
+            'nodeSelector': {
+                'kubernetes.io/hostname': node_hostname
+            }
```

Comment:
This nodeSelector bit is the new part. The rest of this PR is just compulsive refactoring, sorry.